lib/crowbar/client/command/proposal/edit.rb

```diff
@@ -140,6 +150,34 @@ def payload_content
               from_editor
             end
           end
+
+          def check_nodes(configuration)
+            invalid_nodes = []
+
+            configuration["deployment"].each do |service_name, service|
+              next if service["elements"].nil?
+
+              service["elements"].each do |role, nodes|
+                nodes.each do |node|
+                  unless valid_elements.include?(node)
```

Comment:
note that `valid_elements` fetches the list of nodes and clusters every time you run it so it would be better to call it once before the loop and store it in some variable